sql/expression_value.h

```diff
@@ -344,6 +344,12 @@ struct ExpressionValueInfo {
         return true;
     }
 
+    // Does it always returns the same value
+    virtual bool isConst() const = 0;
+
+    //return an expression value info of the same type, with the assigned constness
+    virtual std::shared_ptr<ExpressionValueInfo> getConst(bool constant) const = 0;
```

Comment:
I still don't fully understand why we need a clone and why the constness should not be evaluated and set when the expression value info is created but I don't want to hold the PR for that.  We might want to rename `getConst` to `cloneWithConst` or simply `clone` to avoid the confusion with `isConst`.